src/UI/V2/Dialog.elm

```diff
@@ -0,0 +1,119 @@
+module UI.V2.Dialog exposing
+    ( Dialog(..)
+    , dialogMap
+    , renderElement
+    , withBody
+    , withButtons
+    )
+
+import Element exposing (Element, fill, shrink)
+import Element.Border as Border
+import UI.Button as Button exposing (Button)
+import UI.Icon as Icon exposing (Icon)
+import UI.Internal.Colors exposing (mainBackground)
+import UI.Palette as Palette
+import UI.RenderConfig exposing (RenderConfig)
+import UI.Text as Text
+import UI.Utils.Element as Element
+
+
+type Dialog msg
+    = Dialog Properties (Options msg)
+
+
+type alias Properties =
+    { title : String
+    , icon : Icon
+    }
+
+
+type alias Options msg =
+    { body : Element msg
+    , buttons : List (Button msg)
+    }
+
+
+dialogMap : (a -> b) -> Dialog a -> Dialog b
+dialogMap applier (Dialog { title, icon } { body, buttons }) =
+    Dialog
+        { title = title
+        , icon = icon
+        }
+        { body = Element.map applier body
+        , buttons = List.map (Button.map applier) buttons
+        }
+
+
+withBody : Element msg -> Dialog msg -> Dialog msg
+withBody body (Dialog props options) =
+    Dialog props { options | body = body }
+
+
+withButtons : List (Button msg) -> Dialog msg -> Dialog msg
+withButtons buttons (Dialog props options) =
+    Dialog props { options | buttons = buttons }
+
+
+renderElement : RenderConfig -> Dialog msg -> Element msg
+renderElement cfg dialog =
+    desktopDialogView cfg dialog
+
+
+desktopDialogView : RenderConfig -> Dialog msg -> Element msg
+desktopDialogView cfg (Dialog { title, icon } { body, buttons }) =
+    Element.column
+        [ Element.width shrink
+        , Element.centerY
+        , Element.centerX
+        , mainBackground
+        , Element.padding 32
+        , Border.rounded 6
+        ]
+        [ desktopHeaderRow cfg title icon
```

Comment:
I think it's good to keep using `Element.above`.
The way you've done here, we can't set the width of the dialog from within the page (by using `Element.above` the dialog width will be the same as the body, and the header will match them Otherwise, if the header is longer than the body, it will be the one setting the dialog's width).